### DISCUSSION TITLE
`act` related warnings appeared after migrating `vitest` to `v2`

### DISCUSSION THREAD

**Discussion by tiagojufr** [vitest-dev/vitest#6167] [Q&A]
Hi guys,

I've been using vitest 1.x in my project for a while. Tried upgrading to `2.0.3` today and when running my tests I see a lot of these messages:

```
Warning: An update to <MyComponentName> inside a test was not wrapped in act(...).
When testing, code that causes React state updates should be wrapped into act(...):
act(() => {
  /* fire events that update state */
});
/* assert on the output */
This ensures that you're testing the behavior the user would see in the browser. Learn more at https://reactjs.org/link/wrap-tests-with-act
    at <MyComponentName> ....
```

I don't see these messages iwhen using vitest v1. I followed the migration guide and I didn't see any references to this.

I'm not saying this is a bug in vitest, as I think my components are actually doing state updates when they render, I'm just curious why I didn't get this message in v1 and what changed for it to appear now. Could this be something worth to add to the migration guide?

**Comment by fabriciofx** [comment]
@tiagojufr Have you solved this problem?

**Reply by tiagojufr** [reply]
@fabriciofx no, unfortunately the project where I had these issues was canceled. In the meantime I've been working on other projects using vitest 3 without facing this issue. I guess you can try to upgrade from vitest 1 to 3 and see if that fixes it...

**Reply by fabriciofx** [reply]
@tiagojufr I'm actually using `vitest` 4 and `vitest-browser-react` 2 as you can see [here](https://github.com/fabriciofx/system-bank-react). My last resource is ignore these warning messages in log (I did it [here](https://github.com/fabriciofx/system-bank-react/blob/main/src/components/cliente/ListaClientes.test.tsx))

**Reply by tiagojufr** [reply]
@fabriciofx then those warnings might actually be correct and you may need to wrap the simulated user actions in your tests with `act`. If your components deal with promises, intervals or timeouts that is somehow normal as a re-render will happen when vitest is not expecting it

**Reply by fabriciofx** [reply]
@tiagojufr I have tried putting `act` everywhere you can imagine! I also tried using `waitFor`, but nothing solved the problem.
